### PR TITLE
fix quarantine for live mode (closes #4093)

### LINKED ISSUE #4093: Quarantine and live modes are not working nicely together
### What is the Current behavior?
In both quarantine and live modes failed test will stuck on the first try
 
### What is the Expected behavior?
Failed test should rerun 3 times and print an error of the last run
 
### What is your web application and your TestCafe test code?
```
import { Selector } from 'testcafe';

fixture('Test').page('http://localhost:3000/');

test('Test', async t => {
  console.log('Inside test');
  await t.click(Selector('.test'));
});

```
### Steps to Reproduce:
Clone this repo https://github.com/valerii-cognite/testcafe-quarantine-issue

run `yarn && yarn start`

In order to see a problem run 

`yarn testcafe chrome testcafe --live --quarantine-mode` 

and see theat failed test will never be retriggered and no error will be printed.

In order to see an expected behaviour run 

`yarn testcafe chrome testcafe --live` 

or 

`yarn testcafe chrome testcafe --quarantine-mode` 

and observe an error.

### Your Environment details:
 
* testcafe version:                   1.3.3
* node.js version:                    v8.13.0
* browser name and version: Chrome 75
* platform and version:           macOS 10.13

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/live/test-run-controller.js [hunk 1]
```diff
@@ -81,25 +80,28 @@ class LiveModeTestRunController extends EventEmitter {
     _onTestRunDone (testRun) {
         testRun.state = TEST_RUN_STATE.done;
 
-        const hasRunningTests = this._getTestRuns().some(t => t.state !== TEST_RUN_STATE.done);
+        const testWillBeRestarted            = !this.isTestFinished(testRun);
```
Comment (Farfurix) [inline src/live/test-run-controller.js]:
```suggestion
        const testWillBeRestarted            = !this._isTestFinished(testRun);
```

File: src/live/test-run-controller.js [hunk 2]
```diff
@@ -108,6 +110,15 @@ class LiveModeTestRunController extends EventEmitter {
     _onTestRunReadyToNext (testRun) {
         testRun.setReadyToNext();
     }
+
+    isTestFinished (testRun) {
```
Comment (Farfurix) [inline src/live/test-run-controller.js]:
```suggestion
    _isTestFinished (testRun) {
```

File: src/runner/test-run-controller.js [hunk 3]
```diff
@@ -32,6 +32,12 @@ class Quarantine {
         return failedThresholdReached || passedThresholdReached;
     }
 
+    isFirstAttemptSuccessfull (extraErrors) {
```
Comment (Farfurix) [inline src/runner/test-run-controller.js]:
```suggestion
    isFirstAttemptSuccessful (extraErrors) {
```

File: src/live/test-run-controller.js [hunk 4]
```diff
@@ -108,6 +110,15 @@ class LiveModeTestRunController extends EventEmitter {
...
+        const { quarantine, errs } = testRun;
+
+        if (!quarantine)
+            return true;
+
+        return quarantine.isFirstAttemptSuccessfull(errs) || quarantine.isThresholdReached(errs);
```
Comment (Farfurix) [inline src/live/test-run-controller.js]:
```suggestion
        return quarantine.isFirstAttemptSuccessful(errs) || quarantine.isThresholdReached(errs);
```

**Comment by AlexKamaev** [comment]
@testcafe-build-bot retest

**Pull request merged**